src/redprl/refiner.fun

```diff
@@ -1177,6 +1177,54 @@ struct
     end
   end
 
+  structure Coe =
+  struct
+    fun Eq alpha jdg =
+      let
+        val _ = RedPrlLog.trace "Coe.Eq"
+        val H >> CJ.EQ ((lhs, rhs), ty) = jdg
+        val Syn.COE {dir=(r0, r'0), ty=(u, ty0u), coercee=m0} = Syn.out lhs
+        val Syn.COE {dir=(r1, r'1), ty=(v, ty1v), coercee=m1} = Syn.out rhs
+        val () = assertParamEq "Coe.Eq source of direction" (r0, r1)
+        val () = assertParamEq "Coe.Eq target of direction" (r'0, r'1)
+
+        val ty01 = substSymbol (r1, u) ty0u
+        val (goalTy1, _) = makeGoal @@ ([],[]) || H >> CJ.EQ_TYPE (ty01, ty)
+
+        val w = alpha 0
+        val ty0w = substSymbol (P.ret w, u) ty0u
+        val ty1w = substSymbol (P.ret w, v) ty1v
+        val (goalTy, _) = makeGoal @@ ([(w, P.DIM)],[]) || H >> CJ.EQ_TYPE (ty0w, ty1w)
+
+        val ty00 = substSymbol (r0, u) ty0u
+        val (goalCoercee0, _) = makeGoal @@ ([],[]) || H >> CJ.MEM (m0, ty00)
+        val ty10 = substSymbol (r0, v) ty1v
+        val (goalCoercee1, _) = makeGoal @@ ([],[]) || H >> CJ.MEM (m1, ty10)
+      in
+        T.empty >: goalTy1 >: goalTy >: goalCoercee0 >: goalCoercee1 #> trivial
+      end
```

Comment:
@jonsterling No, it should be either `r'1` or `r1` but not `r0`. `r'0` is `r` on paper.